Add w32-ime recipe

trueroad opened the pull request:
### Brief summary of what the package does

w32-ime makes Japanese input with Windows IME (Input Method Editor) more convenient.

### Direct link to the package repository

https://github.com/trueroad/w32-ime.el

### Your association with the package

Author

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Thanks for this!  My quick reaction to `w32-ime-wrap-function-to-control-ime` is that it could be fragile.  I'm contriving an example, but if a function called `test-original` exists a priori, then it seems the function `test` will never be wrapped.  Is what you're doing here equivalent to wrapping the function with [`:around` advice](https://www.gnu.org/software/emacs/manual/html_node/elisp/Advising-Functions.html)?

Beyond that, the code looks good.  Just a handful of sharp-quote suggestions:

- w32-ime.el#L246: It's safer to sharp-quote function names; use `#'`
- w32-ime.el#L248: It's safer to sharp-quote function names; use `#'`
- w32-ime.el#L283: It's safer to sharp-quote function names; use `#'`
- w32-ime.el#L259: It's safer to sharp-quote function names; use `#'`
- w32-ime.el#L261: It's safer to sharp-quote function names; use `#'`
- w32-ime.el#L269: It's safer to sharp-quote function names; use `#'`

trueroad commented:
Thank you for reviewing.

We've rewritten `w32-ime-wrap-function-to-control-ime` using `add-advice` with `:around`.
Also, we've fixed sharp-quote function names.

riscy commented:
Thanks very much, looks good.  And, welcome to MELPA. :)

trueroad commented:
Thank you!
